mne/viz/backends/tests/test_renderer.py

```diff
@@ -152,3 +153,25 @@ def test_3d_backend(renderer):
                     focalpoint=center)
     rend.reset_camera()
     rend.show()
+
+
+@skips_if_not_pyvista
+def test_get_3d_backend():
+    """Test get_3d_backend."""
+    orig_backend = mne.viz.backends.renderer.MNE_3D_BACKEND
+
+    # Test the private function first.
+    # Test twice to ensure the first call had no side-effect
+    mne.viz.backends.renderer.MNE_3D_BACKEND = None
+    assert mne.viz._get_3d_backend() is None
+    assert mne.viz._get_3d_backend() is None
+
+    # Now test the public function.
+    mne.viz.backends.renderer.MNE_3D_BACKEND = None
+    assert mne.viz.get_3d_backend() is None
+
+    mne.viz.set_3d_backend('pyvista')
+    assert mne.viz.get_3d_backend() == 'pyvista'
```

Comment:
Your code as-is is a bit unsafe. For example if this `assert` fails, now the 3D backend has been set for any other tests that also run. Any time you modify a global state or var you should enclose it effectively in a `try/except` or equivalent pattern (context manager, Pytest fixture). Here I'd probably use some pytest magic:
```
@pytest.fixture()
def restore_backend():
    import mne
    orig_backend = mne.viz.backends.renderer
    yield
    mne.viz.backends.renderer = orig_backend


def test_get_3d_backend(restore_backend):
    ...
```
Then pytest will ensure that no matter what happens, it will get set back.

Reply:
Very good point, will try to adopt this.

Reply:
Adressed in 2adeb3d. Turns out there was already a fixture I could reuse.